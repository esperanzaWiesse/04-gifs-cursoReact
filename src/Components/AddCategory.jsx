import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };

  const onSubmit = (event) => {
    // preventDefault -> evita que al pesionar enter, la pag se recargue y se pierda el valor del inputValue
    event.preventDefault();

    if (inputValue.trim().length <= 1) return;

    // categories -> reprensenta todas las categorias que ya existen, es decir el estado actual
    // setCategories( categories => [inputValue, ...categories])

    // limpiendo la caja de texto, despues de ingresar un nuevo valor
    setInputValue("");

    onNewCategory(inputValue.trim());
  };

  return (
    <form onSubmit={onSubmit} aria-label="form"> // aria-label="form" para que los test encuentren el form
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
