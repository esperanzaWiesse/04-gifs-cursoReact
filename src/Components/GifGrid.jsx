import { GitItem } from "./GitItem";
import { useFetchGifs } from "../hooks/useFetchGifs";


export const GifGrid = ({category}) => {

  const {images, isLoading} = useFetchGifs(category);

  console.log({images, isLoading});

  return (
    <>
        <h3>{category}</h3>
        {/* el elemento html h2 no debe de existir a menos que se se requira mostrarlo*/}
        {/* forma 2 -> (forma recomendada) (es la implementada)*/}
        { isLoading && (<h2>Cargando ....</h2>)}


        {/* forma 1 -> usando un ternario */}
        {/* { isLoading ? (<h2>Cargando ....</h2>): null } */}
        {/* forms 3 -> creando un componente que tenga la lojica y se pueda usar en varias partes*/}
        {/* { < isLoading isLoading={isLoading}></isLoading>} */}
        
        <div className="card-grid">
          {
            images.map((images) => (
              <GitItem 
                key={images.id}
                // mandar asi las props, se llama exparsir las props, lo que hace es devolver un arreglo de componentes GitItem [GitItem, GitItem, GitItem, GitItem]; donde cada componente de muestro arreglo correponde a cada elemento del arregli de gifts
                {...images}
              />
            ))
          }
        </div>
    </>
  )
}
