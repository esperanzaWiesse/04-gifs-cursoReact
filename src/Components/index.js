
// el arterisco indica que va exportar todo lo que el componente experta 
export * from "./AddCategory";
export * from './GifGrid';
export * from './GitItem';

