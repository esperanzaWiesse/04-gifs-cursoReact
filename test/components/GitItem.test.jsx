import { render, screen } from "@testing-library/react"
import { GitItem } from "../../src/Components/GitItem";

describe('Test del componente GitItem', () => {

    const title = 'sakura';
    const url = 'http://sakura.com/kawai';

    test('debe de hacer match con el snapshot', () => { 

        const {container} = render(<GitItem title={title} url={url}/>);
        expect(container).toMatchSnapshot();

     })

     test('debe de mostrar la imagen con el URL y el ALT indicado ', () => { 
        
        render(<GitItem title={title} url={url}/>);
        // screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);     -> esta es otra forma de hacer el test 
        const { src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(alt);
      })

      test('debe de mostrar el titulo en el componente', () => { 
        render(<GitItem title={title} url={url}/>);
        expect(screen.getByText(title)).toBeTruthy();
       })

 })
