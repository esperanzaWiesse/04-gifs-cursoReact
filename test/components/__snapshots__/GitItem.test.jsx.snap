// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test del componente GitItem debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="sakura"
      src="http://sakura.com/kawai"
    />
    <p>
      sakura
    </p>
  </div>
</div>
`;
